File: benchmarks/poly/regex/regex.py
import re
import sys, time


def main():
    _t0 = time.perf_counter_ns()
    n = 200000
    pattern = re.compile(r"user(\d+) GET /api/(\d+) (\d+) (\d+)ms")
    checksum = 0
    match_count = 0
    for i in range(n):
        status = 200 if i % 5 != 4 else 404
        line = f"user{i} GET /api/{i % 50} {status} {i % 250}ms"
        m = pattern.match(line)
        if m:
            match_count += 1
            uid = int(m.group(1))
            path = int(m.group(2))
            st = int(m.group(3))
            ms = int(m.group(4))
            checksum = (checksum * 31 + uid + path + st + ms) % 1000000007

    print(match_count)
    print(checksum)
    print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)


if __name__ == "__main__":
    main()

File: benchmarks/poly/regex/regex.lua
local function main()
    local _t0 = os.clock()
    local n = 200000
    local checksum = 0
    local match_count = 0

    for i = 0, n - 1 do
        local status = (i % 5 ~= 4) and 200 or 404
        local line = "user" .. i .. " GET /api/" .. (i % 50) .. " " .. status .. " " .. (i % 250) .. "ms"

        local uid, path, st, ms = line:match("^user(%d+) GET /api/(%d+) (%d+) (%d+)ms$")
        if uid then
            match_count = match_count + 1
            checksum = (checksum * 31 + tonumber(uid) + tonumber(path) + tonumber(st) + tonumber(ms)) % 1000000007
        end
    end

    print(match_count)
    print(checksum)
    io.stderr:write(string.format("bench_ns=%.0f\n", (os.clock() - _t0) * 1e9))
end

main()
